src/org/openlcb/implementations/DatagramMeteringBuffer.java

```diff
@@ -267,8 +280,23 @@ public void handleDatagramRejected(DatagramRejectedMessage msg, Connection sende
      */
     public void dispose(){
         // shut down the thread pool
-        if(threadPool != null) {
-           threadPool.shutdownNow();
+        if(threadPool != null && !(threadPool.isShutdown())) {
```

Comment:
We need to add a check here that we are only shutting down threadpools that we created ourselves. We should not shut down a threadpool that was passed in.